include/CL/sycl/buffer/detail/buffer.hpp

```diff
@@ -219,18 +318,64 @@ class buffer : public detail::buffer_base,
   }
 
 
-  /** Set the weak pointer to copy back data on buffer deletion
+  /** Set the weak pointer as destination for write-back on buffer destruction.
+  */
+  void set_final_data(std::weak_ptr<T> && final_data) {
+    shared_pointer = boost::none;
```

Comment:
No, you cannot erase this because it holds the original buffer shared pointer at construction.